### DISCUSSION TITLE
Lifebar messages feedback (action.zss)

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1017] [Other] [answered]
There are a few odd things with the default lifebar messages:

1. Helpers don't show counter hit messages

2. The Reversal message refers to using a ReversalDef. Normally a Reversal is when you perform a special or super move (or some other actions) in the very first frame after coming out of Movetype = H (or getting up, etc)

3. The Parry message could maybe be somehow expanded to not refer specifically to my Parry code. I think that breaks the universality of a default code

Regarding 2 and 3, after fixing the Reversal message, you could just make Parry be displayed when any ReversalDef is successful.

**Comment by K4thos** [comment, marked as answer]
as always we would appreciate PR with those changes, if you know how to implement them. No idea how the second point should be coded.

**Comment by potsmugen** [comment]
Yeah, the second one is tricky as it'd rely on checking common states for specials and supers (1000-4999), which is not perfect. Elecbyte recommended those numbers and most people follow them, but still.
It'd also ideally trigger on time = 0 of those states, which isn't doable.

**Comment by potsmugen** [comment]
Forgot to mention but I implemented points 2 and 3 in the default files, if anyone wants to try them.

Point 1 I wasn't happy with how it turned out, because allowing it for helpers makes it so that if two helper fireballs collide with each other, a counter hit message will pop up. Maybe it can be revisited as new triggers get added to the engine.

**Comment by potsmugen** [comment]
I guess I can leave this here instead of making a new report. Some new issues popped up recently with the refactor:

- The Reversal and Just Defended messages seem to have stopped working altogether

- Messages as a whole lost the ignorehitpause property (although it only matters for some)

- If two helper fireballs collide with each other, it will negate the first attack message. I think the first attack trigger should only be checked when hitting the root players (or maybe also player type helpers)

- I think the functions could have some system prefix. I can see people making their own "CounterHit" etc functions and being puzzled when they cause issues
